packages/backend/ubuntu-pro.go

```diff
@@ -0,0 +1,256 @@
+package main
+import (
+    "os/exec"
+    "log"
+    "context"
+    "errors"
+    pb "github.com/canonical/desktop-security-center/packages/proto"
+    epb "google.golang.org/protobuf/types/known/emptypb"
+    wpb "google.golang.org/protobuf/types/known/wrapperspb"
+    "github.com/tidwall/gjson"
+    "github.com/godbus/dbus/v5"
+    "google.golang.org/grpc/codes"
+    "google.golang.org/grpc/status"
+)
+
+type ProServer struct {
+    pb.UnimplementedProServer
+    conn             *dbus.Conn
+    apps             dbus.BusObject
+    infra            dbus.BusObject
+    livepatch        dbus.BusObject
+    manager          dbus.BusObject
+    reqId            string
+}
+
+func NewProServer(conn *dbus.Conn) (*ProServer, error) {
+    s := &ProServer{
+        conn: conn,
+    }
+
+    s.infra = conn.Object(
+        "com.canonical.UbuntuAdvantage",
+        "/com/canonical/UbuntuAdvantage/Services/esm_2dinfra",
+    )
+    err := s.infra.Call("org.freedesktop.DBus.Peer.Ping", 0).Err
+    if err != nil {
+        return nil, status.Errorf(codes.Internal, "failed to ping DBus ESM-Infra object")
+    }
+
+    s.apps = conn.Object(
+        "com.canonical.UbuntuAdvantage",
+        "/com/canonical/UbuntuAdvantage/Services/esm_2dapps",
+    )
+    err = s.apps.Call("org.freedesktop.DBus.Peer.Ping", 0).Err
+    if err != nil {
+        return nil, status.Errorf(codes.Internal, "failed to ping DBus ESM-Apps object")
+    }
+
+    s.livepatch = conn.Object(
+        "com.canonical.UbuntuAdvantage",
+        "/com/canonical/UbuntuAdvantage/Services/livepatch",
+    )
+    err = s.livepatch.Call("org.freedesktop.DBus.Peer.Ping", 0).Err
+    if err != nil {
+        return nil, status.Errorf(codes.Internal, "failed to ping DBus ESM-Apps object")
+    }
+
+    s.manager = conn.Object(
+        "com.canonical.UbuntuAdvantage",
+        "/com/canonical/UbuntuAdvantage/Manager",
+    )
+    err = s.manager.Call("org.freedesktop.DBus.Peer.Ping", 0).Err
+    if err != nil {
+        return nil, status.Errorf(codes.Internal, "failed to ping DBus Manager object")
+    }
+
+    return s, nil
+}
+
+func isServiceEnabled(obj dbus.BusObject) (bool, error) {
+    status, err := obj.GetProperty("com.canonical.UbuntuAdvantage.Service.Status")
+    if err != nil {
+        log.Println(obj.Path(), err)
+        return false, err
+    }
+    return status.Value().(string) == "enabled", nil
+}
+
+/* Determines if system is attached to Ubuntu Pro. */
+func (s *ProServer) IsMachineProAttached(ctx context.Context, _ *epb.Empty) (*wpb.BoolValue, error) {
+    isAttached, err := s.manager.GetProperty("com.canonical.UbuntuAdvantage.Manager.Attached")
+    if err != nil {
+        log.Println(err)
```

Comment:
Nothing wrong with go's log package, but we prefer [log/slog](https://pkg.go.dev/log/slog), as it allows for richer log messaging. You don't have to fix it now, you can look into this for another PR, just want to let you know!

Allows you to differentiate between:
```
 slog.Error(fmt.Sprintf("somthing went wrong: %v", err))
 slog.Info("somthing happened!")
 slog.Warn("somthing concerning happened!")
 slog.Debug("somthing a dev might be interested in happened!")
```
The gotcha of slog is that it doesn't currently support formatting strings out the box, so you need to wrap errors in a `fmt.Sprintf(...)`, but you do end up with much nicer logs:

```
2024/02/09 09:41:34 INFO Using configuration file: example.config.yaml
2024/02/09 09:41:35 INFO Serving GRPC requests on /home/matt/test.sock
```

Reply:
Ack! I will address this in a later merge request then.